Update README.rst (#683)
mongodb/specifications: source/server-discovery-and-monitoring/tests/README.rst

A "server" object represents a correct ServerDescription within the client's
current TopologyDescription. It has the following keys:

> - type: A ServerType name, like "RSSecondary". For multi-threaded implementations, the "PossiblePrimary" type should be interpreted as "RSUnknown".

Reviewer: I'll also point out that this section is mainly a description of the test format, and isn't the ideal place to include implementation details about SDAM itself.

Reviewer 2: I think this is a useful clarification of the test format itself.

Author: Yeah, when implementing it I was unclear whether I should ignore the cases with PossiblePrimary.

Reviewer: I'd suggest adding the following to the end of the sentence:

```
(see: [ServerType](../server-discovery-and-monitoring.rst#servertype))
```

Cross-referencing it will at least point back to the canonical source of info for this. Also, it may be helpful to refer to "async and multi-threaded implementations".

Reviewer 3: @lamontnelson do you have an opinion on @jmikola s comment?

Author: Sorry, this seems to have slipped through the cracks. I've added "async and multi-threaded implementations" and linked to the specification.

> - type: A ServerType name, like "RSSecondary". For async and multi-threaded implementations, the "PossiblePrimary" type should be interpreted as "RSUnknown". See `ServerType <../server-discovery-and-monitoring.rst#servertype>`_ for more details.

Reviewer: Suggestion per @p-mongo's comment in https://github.com/mongodb/specifications/pull/683#issuecomment-550047807.

```suggestion
- type: A ServerType name, like "RSSecondary". See `ServerType <../server-discovery-and-monitoring.rst#servertype>`_ for details pertaining to async and multi-threaded drivers.
```

I realize this PR originally used "RSUnknown", but I only see "Unknown" used in the spec itself. I'm not sure if that changed in the time since this PR was created, but if so it's probably all the more reason to refer back to the spec instead of duplicating spec language in the test README.

Author: Thanks, I like this change. This should be ready to go now. Better late than never!
